### PR TITLE
Refactor of command-line-data-tools

### PR DESCRIPTION
**Pull request by pazaan**
* Move common functions to JavaScript library
* Add linting
* Tidy up so that we can publish to `npm`

### CODE REVIEW

File: bin/getdata.js [hunk 1]
```diff
@@ -1,112 +1,104 @@
...
-						}).pipe(outstream);
-
-					});
-			});
-
-	})
-	.parse(process.argv);
+const program = require('commander');
+const prompt = require('prompt-sync')();
+const rp = require('request-promise-native');
```
Comment (jh-bate) [inline bin/getdata.js]: `rp`  maybe just `requestPromise` for readability? I only say that as I had to scroll to the top to figure out what it was

File: lib.js [hunk 2]
```diff
@@ -0,0 +1,1861 @@
+/* eslint-disable no-param-reassign, no-underscore-dangle */
+
+const crypto = require('crypto');
+
+/* Sort Data */
+function sortData(data) {
```
Comment (jh-bate) [inline lib.js]: `sortDataByDate` ??

File: package.json [hunk 3]
```diff
@@ -1,10 +1,12 @@
 {
-  "name": "tidepool-command-line-data-tools",
+  "name": "tidepool-data-tools",
   "version": "0.0.1",
```
Comment (jh-bate) [inline package.json]: version bump??

File: bin/scoredata.js [hunk 4]
```diff
@@ -1,290 +1,274 @@
...
+
+const program = require('commander');
+const Excel = require('exceljs');
+const fs = require('fs');
+const chalk = require('chalk');
+const ss = require('simple-statistics');
+const path = require('path');
 
 const MAX_EGV_PER_DAY = 288;
 const DAY_IN_MILLI = 86400000;
```
Comment (jh-bate) [inline bin/scoredata.js]: this is also exported in `lib` if you want to rationalise

File: bin/getdata.js [hunk 5]
```diff
@@ -1,112 +1,104 @@
...
-								console.log(chalk.green.bold('Successful data request.'));
-							}
-						}).pipe(outstream);
-
-					});
-			});
-
-	})
-	.parse(process.argv);
+const program = require('commander');
```
[the remark this answers is not in the document]
Comment (pazaan) [reply bin/getdata.js]: `import` is still unstable in node. So to support it, we'd need to use Babel.

File: bin/scoredata.js (hunk 4 above)
Comment (pazaan) [reply bin/scoredata.js]: They're not in `lib`. They're command-line imports, so they're fine here. We wouldn't want to import them for JS-only use.

**Pull request merged**